numbat/src/lib.rs

```diff
@@ -394,6 +418,57 @@ impl Context {
             return help;
         }
 
+        if let Some((_, fn_metadata)) = self.interpreter.lookup_function(keyword) {
+            let metadata = fn_metadata.clone();
+
+            let mut help = m::text("Function: ");
+            if let Some(name) = &metadata.name {
+                help += m::text(name);
+            } else {
+                help += m::identifier(keyword);
+            }
+            if let Some(url) = &metadata.url {
+                help += m::text(" (") + m::string(url_encode(url)) + m::text(")");
+            }
+            help += m::nl();
+
+            if let Ok((statements, _)) = self.interpret(keyword, CodeSource::Internal) {
+                let signature = match Statement::as_expression(&statements[0]) {
+                    Some(expression) => expression
+                        .get_type()
+                        .to_readable_type(self.dimension_registry()),
+                    None => m::empty(),
+                };
+                help += m::text("Signature: ") + signature + m::nl();
+            }
```

Comment:
This is a nice hack, but it might be better/easier to add a querying function to `TypeChecker` which has all the information we need (in `TypeChecker::function_signatures`). That could be called here using something like `self.typechecker.get_function_signature(keyword)`. But then you would need to turn `FunctionSignature` into something that you can render, so… let's maybe stick with the current solution for now. Things are about to change significantly in the type checker, which might also affect how function signatures are stored.

Reply:
Yes, I was not really happy about this part of the code either. I will see if I can implement a method to query and print `TypeChecker::function_signatures`, thanks for that hint!

Reply:
I rewrote this part, the function signature is now directly extracted from `TypeChecker::FunctionSignature` and now also works for generic functions.